src/external_integrations/philips_hue_test.py

```diff
@@ -52,57 +68,96 @@ def test_set_color(philips_hue_client, requests_mock):
     assert "{'success': {'/lights/1/state/hue': '0'}}" in response.text
 
 
-def test_trigger_from_incident_invalid_state(philips_hue_client, requests_mock):
-    notification = {"incident": {"state": "unknown"}}
+def test_trigger_from_incident_bad_url(philips_hue_client, requests_mock):
+    bridge_ip_address = philips_hue_client.bridge_ip_address
+    username = philips_hue_client.username
+    matcher = re.compile(f'http://{bridge_ip_address}/api/{username}')
+    requests_mock.register_uri('PUT', matcher,
+                               text=philips_hue_mock.mock_hue_put_response)
+
+    with pytest.raises(philips_hue.BadAPIRequestError) as e:
+        assert philips_hue_client.set_color('2', 65535)
+    assert str(e.value) == 'invalid Philips Hue url'
+
+
+def test_trigger_from_incident_invalid_state(config, philips_hue_client,
+                                             requests_mock):
+    notification = {'incident': {'policy_name': 'policyB', 'state': 'unknown'}}
     bridge_ip_address = philips_hue_client.bridge_ip_address
     username = philips_hue_client.username
     matcher = re.compile(f'http://{bridge_ip_address}/api/{username}')
     requests_mock.register_uri('PUT', matcher,
                                text=philips_hue_mock.mock_hue_put_response)
 
     with pytest.raises(philips_hue.UnknownIncidentStateError) as e:
-        assert philips_hue.trigger_light_from_monitoring_notification(
-            philips_hue_client, notification, 1)
+        assert philips_hue.get_target_hue_from_monitoring_notification(
+            notification, config["POLICY_HUE_MAPPING"])
     assert 'Incident state must be "open" or "closed"' in str(e.value)
 
 
-def test_trigger_from_incident_bad_url(philips_hue_client, requests_mock):
-    notification = {"incident": {"state": "open"}}
+def test_trigger_hue_with_nondefault_open_incident(config, philips_hue_client,
+                                                   requests_mock):
+    policy_name = 'policyB'
+    state = 'open'
+    notification = {'incident': {'policy_name': policy_name, 'state': state}}
     bridge_ip_address = philips_hue_client.bridge_ip_address
     username = philips_hue_client.username
     matcher = re.compile(f'http://{bridge_ip_address}/api/{username}')
     requests_mock.register_uri('PUT', matcher,
                                text=philips_hue_mock.mock_hue_put_response)
 
-    with pytest.raises(philips_hue.BadAPIRequestError) as e:
-        assert philips_hue.trigger_light_from_monitoring_notification(
-            philips_hue_client, notification, 2)
-    assert str(e.value) == 'invalid Philips Hue url'
+    response = philips_hue.get_target_hue_from_monitoring_notification(
+        notification, config['POLICY_HUE_MAPPING'])
+
+    assert response == config['POLICY_HUE_MAPPING'][policy_name][state]
+
+
+def test_trigger_hue_with_nondefault_closed_incident(config, philips_hue_client,
```

Comment:
Two questions with these tests in general:

1. What is the purpose of the mocking? I don't think that `get_target_hue_from_monitoring_notification()` makes an HTTP call that we need to mock out? Or am I misreading?
1. The variable is called `response`, isn't that actually a `hue_value`?
1. In general, tests should read like a story. Just looking at the test itself, it should be plainly obvious that the test is correct (since we don't test our tests). So, in general, inline statements should be preferred over test fixtures, because the test fixture is an abstraction that makes the testcase less literal and less obvious. Of course, discretion should be applied here, because sometimes it's true that test fixtures actually make the testcase **more** clear.

In this case, however, loading a simple dict into a fixture, just to deference it feels like unnecessary obfuscation. For instance, this test could be rewritten as:
```
policy_name = 'policyA'
incident_state = 'closed'
expected_hue_value = 42237
notification = {'incident': {'policy_name': policy_name, 'state': incident_state}}
policy_hue_mapping = {
        policy_name: {
            'open': 5620,
            incident_state: expected_hue_value
        },
        'default': {
            'open': 65280,
            'closed': 24432
        }
    }

actual_hue_value = philips_hue.get_target_hue_from_monitoring_notification(notification, policy_hue_mapping)

assert  actual_hue_value == expect_hue_value

```
I think most of the documentation on this is internal, unfortunately.

Also helpful, prefer DAMP over DRY: https://stackoverflow.com/questions/6453235/what-does-damp-not-dry-mean-when-talking-about-unit-tests